src/arrayMethodJoin.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinStr = '';
+    let tempSeparator = separator;
+
+    if (tempSeparator === undefined) {
+      tempSeparator = ',';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      const checkStr = this[i];
+
+      if (checkStr === null || checkStr === undefined) {
+        joinStr += '';
+      } else if (checkStr === 'null') {
+        joinStr += 'null';
+      } else {
```

Comment:
Else if condition will always be false. You need to alter the logic.
if `checkStr === null` is true - else if will never execute.